app/src/main/java/org/onionshare/android/ui/share/ShareUi.kt

```diff
@@ -71,23 +69,23 @@ private val bottomSheetPeekHeight = 60.dp
 @OptIn(ExperimentalMaterialApi::class)
 fun ShareUi(
     navController: NavHostController,
-    stateFlow: StateFlow<ShareUiState>,
+    shareState: ShareUiState,
+    filesState: FilesState,
     onFabClicked: () -> Unit,
     onFileRemove: (SendFile) -> Unit,
     onRemoveAll: () -> Unit,
     onSheetButtonClicked: () -> Unit,
 ) {
-    val state = stateFlow.collectAsState()
     val scaffoldState = rememberBottomSheetScaffoldState()
     val offset = getOffsetInDp(scaffoldState.bottomSheetState)
-    if (state.value == ShareUiState.NoFiles) {
+    if (shareState == ShareUiState.AddingFiles && filesState.files.isEmpty()) {
```

Comment:
Seems like maybe the NoFiles state was useful here?

Reply:
Right, here it would be useful. But separating the `shareState` from the `fileState` is an improvement in the current approach otherwise, I'd argue.